site/src/pages/DeploySettingsPage/OAuth2AppsSettingsPage/CreateOAuth2AppPage.tsx

```diff
@@ -0,0 +1,42 @@
+import { useMutation } from "react-query";
+import { postOAuth2App } from "api/api";
+import type * as TypesGen from "api/typesGenerated";
+import { displayError, displaySuccess } from "components/GlobalSnackbar/utils";
+import { FC } from "react";
+import { useNavigate } from "react-router-dom";
+import { CreateOAuth2AppPageView } from "./CreateOAuth2AppPageView";
+import { pageTitle } from "utils/page";
+import { Helmet } from "react-helmet-async";
+
+const CreateOAuth2AppPage: FC = () => {
+  const navigate = useNavigate();
+
+  const postMutation = useMutation({
+    mutationFn: postOAuth2App,
+    onSuccess: (newApp: TypesGen.OAuth2App) => {
```

Comment:
1. Oh yeah I noticed that too but it felt weird to have the `displaySuccess` stuff in there, however I just realized I can just `await` the mutate and then still do the display in here, I think?  Will fix!
2. Will do!  I was actually not sure if I needed to here because of the navigation; when I navigated back it seemed to have the updated values but is there maybe a case where it might still use the old values?  Even if not necessary, invalidating seems more correct anyway.

Reply:
Huh actually sometimes the icon is missing if I add it and navigate back, probably due to missing invalidation.